projects/miragesdk/src/Makefile

```diff
@@ -1,39 +1,25 @@
-BASE=ocaml/opam:alpine-3.5_ocaml-4.04.0
-
-FILES=$(shell find . -name jbuild) \
-	$(shell find sdk/ -name '*.ml') \
-	$(shell find sdk/ -name '*.mli') \
-	dhcp-client/bpf/dhcp.c dhcp-client/main.ml \
-	dhcp-client-calf/unikernel.ml dhcp-client-calf/config.json
-
-IMAGE=dhcp-client
-
+.PHONY: tag push
 default: push
-	@
 
-.build: Dockerfile $(FILES)
-	docker build $(NO_CACHE) -t $(IMAGE) -f Dockerfile .
-	docker build $(NO_CACHE) -t $(IMAGE) -f Dockerfile -q . > .build || \
-	 (rm -f $@ && exit 1)
+ORG?=miragesdk
+IMAGE=dhcp-client
+NOCACHE?=
 
-hash: Makefile Dockerfile $(FILES) .build
-	{ cat $^; \
-	  docker run --rm --entrypoint sh $(IMAGE) -c 'cat /lib/apk/db/installed'; \
-	  docker run --rm --entrypoint sh $(IMAGE) -c 'opam list'; } \
-	| sha1sum | sed 's/ .*//' > $@
+DEPS=Dockerfile \
+	$(shell find . -name jbuild) \
+	$(shell find . -name '*.ml') \
+	$(shell find . -name '*.mli') \
+	$(shell find . -name '*.c') \
+	$(shell find . -name '*.json')
 
-tag: .build
-	docker tag $(IMAGE) linuxkitprojects/$(IMAGE):$(shell cat hash)
+HASH?=$(shell git ls-tree HEAD -- ../$(notdir $(CURDIR)) | awk '{print $$3}')
```

Comment:
strictly speaking, this is not correct as the `Dockerfile` seems to pull in things like `alpine:3.5` from the network so the package build is not entirely dependent on the content of the directory. This is more like an FYI, ie something which needs addressing at some point